packages/medusa/src/api/middlewares/invalidate-cached-events.ts

```diff
@@ -0,0 +1,23 @@
+import { NextFunction, Request, Response } from "express"
+import { isDefined, MedusaError } from "medusa-core-utils"
+import { IEventBusService } from "../../interfaces/services/event-bus"
+
+export default () => {
+  return (
+    err: MedusaError,
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ) => {
+    const cacheKey = req.request_context.cache_key
```

Comment:
question: request_context is an object pass to the authorize payment, session, cart completion etc, should the property be stored either in the header or directly on the request? Or is that fine?